### PR TITLE
Reject orphaned > and } in XJSText

### LINKED ISSUE #34: Better error messages for unlcosed XJS tags
https://github.com/facebook/react/issues/353 really belongs here since this is a parse error coming from esprima.

### PR DESCRIPTION
**Pull request by syranide**
Stricter parsing to prevent mistakes from throwing very unintuitive errors, fixes #34 and facebook/react#353. It's also consistent with orphaned `<` and `{` (eventually throwing errors) rather than being interpreted as text.

Test plan: `npm test`

### CODE REVIEW

**Comment by syranide** [comment]
cc @jeffmo @benjamn

File: test/fbtest.js [hunk 1]
```diff
@@ -2781,6 +2470,20 @@ var fbTestFixture = {
             message: "Error: Line 1: Unexpected token ILLEGAL"
         },
 
+        '<a>></a>': {
+            index: 2,
+            lineNumber: 1,
+            column: 3,
+            message: "Error: Line 1: Unexpected token >>"
```
Comment (syranide) [inline test/fbtest.js]: ~~I thought about changing this to say just `>` (it's easy) as one would expect it to say, but I imagine this is actually highlights the mistake more clearly and may be preferable.~~ Changed by #37 and will say only `>`.

**Comment by syranide** [comment]
Ping

**Comment by syranide** [comment]
Ping? :)

**Comment by syranide** [comment]
@jeffmo Rebased.

**Comment by jeffmo** [comment]
Before merging this, let's have a codemod script ready to go for it.
(Either in this diff or another works...)

**Comment by syranide** [comment]
I'll look into a codemod as well.

**Comment by syranide** [comment]
Codemod https://github.com/facebook/react/pull/3020

I'll rebase this PR tomorrow, unable to access my workstation for now :/

**Comment by jeffmo** [comment]
cc @gabelevi

**Comment by syranide** [comment]
Rebased... wouldn't it make more sense not to keep `fbtest.rec.js` in the repo?

**Comment by jeffmo** [comment]
No, the main point of `fbtest.rec.js` is to see the differences of the output in the pull requests/diffs (while making it less of a pain to have to copy/paste them inline next to the test string yourself)

**Comment by jeffmo** [comment]
Thanks again, @syranide !

**Comment by syranide** [comment]
@jeffmo Ah yes that makes sense. :+1:

**Pull request merged**